packages/app-store/routing-forms/lib/evaluateRaqbLogic.ts

```diff
@@ -0,0 +1,40 @@
+"use client";
+
+import { Utils as QbUtils, type JsonTree } from "react-awesome-query-builder";
+import jsonLogic from "./jsonLogic";
+import { safeStringify } from "@calcom/lib/safeStringify";
+export const enum RaqbLogicResult {
+  MATCH = "MATCH",
+  NO_MATCH = "NO_MATCH",
+  LOGIC_NOT_FOUND_SO_MATCHED = "LOGIC_NOT_FOUND_SO_MATCHED",
+}
+
+export const evaluateRaqbLogic = ({
+  queryValue,
+  queryBuilderConfig,
+  data,
+  beStrictWithEmptyLogic = false,
+}: {
+  queryValue: JsonTree;
+  queryBuilderConfig: any;
+  data: Record<string, unknown>;
+  beStrictWithEmptyLogic?: boolean;
+}): RaqbLogicResult => {
+  const state = {
+    tree: QbUtils.checkTree(QbUtils.loadTree(queryValue), queryBuilderConfig),
+    config: queryBuilderConfig,
+  };
+  const jsonLogicQuery = QbUtils.jsonLogicFormat(state.tree, state.config);
+  const logic = jsonLogicQuery.logic;
+  if (!logic) {
+    if (beStrictWithEmptyLogic && queryValue.children1 && Object.keys(queryValue.children1).length > 0) {
```

Comment:
We could use this stricter approach with newly introduced attributes. Using it in existing FormFields query could cause possible crashes.

It allows to crash in case of faulty logic instead of just considering it true always.